README.md

```diff
@@ -6,11 +6,11 @@ Status: stage 2
+In WebIDL, potentially ill-formed Strings are referred to using the [DOMString](https://webidl.spec.whatwg.org/#idl-DOMString) type. Well-formed strings are referred to using the [USVString](https://webidl.spec.whatwg.org/#idl-USVString) type. Due to their asymmetric value spaces, conversions from `DOMString` to `USVString` are lossy, whereas `DOMString` can represent any `USVString`. Upon conversion from `DOMString` to `USVString`, common options are to replace unpaired surrogates with the replacement character (`U+FFFD`) or to throw an error. In mixed systems, [WTF-8](https://simonsapin.github.io/wtf-8/) is often utilized to mitigate issues from too early sanitization.
 
-The WebAssembly [Component Model](https://github.com/WebAssembly/component-model) requires well-formed strings, as do some compile-to-JS programming languages, many data encodings, network interfaces, filesystem interfaces, etc. Interfacing JavaScript strings with such APIs is a common use case that therefore suffers from conversion burdens. In particular because conversion from `DOMString` to `USVString` is lossy (common options are to replace unpaired surrogates or to throw an error) there is a regular need for string validation both within the platform and for certain userland use case scenarios.
+The authors of the WebAssembly [Component Model](https://github.com/WebAssembly/component-model) have, as part of the WebAssembly CG's decision-making process to advance the Component Model [to Phase 1](https://github.com/WebAssembly/proposals), achieved consensus under notable opposition (including a [formal objection](https://www.assemblyscript.org/standards-objections.html#component-model-2022-09)) to only support `USVString` on WebAssembly component boundaries. The authors of the Component Model consider Unicode text to be the best design for their users, whereas the concern expressed is that support for Java-like languages and safely interfacing with JavaScript and Web APIs is not covered by the Component Model. The authors of the Component Model have chosen the replacement strategy, so interfacing `DOMString`s with WebAssembly components, including their envisioned ESM-integration, will be subject to silent mutation of string data. Independently, there is a significant number of data encodings, network interfaces, filesystem interfaces, compile-to-JS programming languages, etc. designed for Unicode text (`USVString`) rather than `DOMString`. Therefore, interfacing JavaScript strings with such APIs is, also given the preferences of the WebAssembly CG, an increasingly common use case that suffers from the outlined conversion burdens. To detect respectively address the side-effect of silent data mutation or thrown errors early, there is a regular need for manual string validation and sanitization both within the platform and for certain userland use case scenarios. This proposal addressses both detection and sanitization.
```

Comment:
Why do the minutiae of the wasm descision-making process matter here? It’s clear that you object to this decision in wasm but readers here shouldn’t need to know or care that anyone objected.

Reply:
I think the context is important here, because, since it is directly connected, it increases the surface area this proposal has to cover, and in part motivates the timing to pursue the proposal at this point in time. In particular, if the Component Model would revise its decision and allow passing `DOMString`s (that are also Java, Dart, Kotlin, C#, etc. strings) over WebAssembly boundaries (i.e. to each other and JS / Web APIS) without eagerly mutating them, the decision would impact the scope of the problem statement as per the problem statement. One critical aspect is, as I think, that the Component Model is on track to be used for ESM integration (hence mentioning), which would introduce the complications this proposal addresses for the entire platform when JavaScript modules are transparently replaced with WebAssembly modules, either directly or in dependencies. Do you have ideas how to integrate these relevant aspects in a form that you'd be OK with?

Reply:
The usefulness of this proposal exists even if wasm wasn’t part of the discussion.

I think documenting wasm’s decision is sufficient, and zero context about the decision besides *maybe* a link is both sufficient and desirable.

Reply:
While I would agree that the proposal can be considered useful independently, all of timing, reasoning and extent of performance considerations in browser implementations are directly connected to the new situation introduced by the Component Model, as it significantly adds to scope and urgency. To illustrate, I contend that there is an alternative time line where userland implementations would be considered sufficient, say if the WebAssembly CG had more thoroughly addressed the far-reaching concerns and decided differently. The extraordinary platform complications, and why they will exist, certainly deserve a mention in my opinion. Open to restructure if you have ideas how to integrate these important aspects in a form that is more desirable to you.

Reply:
I think that merely stating the wasm semantics with zero of the history is sufficient to convey that.

Reply:
I certainly recognize that there is a tension here, that, as it seems, revolves around a preference that little to zero context is provided. On the one hand, I would argue that a felt sense for zero context has little weight, and on the other that similar lack of context has led to the Component Model becoming unnecessarily suboptimal for many popular programming languages and, here in particular, JS / Web API interop affecting ECMAScript as well, now extending the complications to the entire platform. From what I've witnessed, bad outcomes are best prevented by providing related context, better more than too little, so interested parties are well-informed before decisions are made. And in this case, the context helps to answer important questions: "Why even?" and "Why now?"

Reply:
The issue is that the "why" of WASM's decision *does not matter here*. TC39 proposal readmes aren't a pulpit for WASM opinions :-)

WASM decided. Therefore, that's the world we live in, whether it's better or worse is irrelevant. The extra (quite editorial content, because "neutral" is an editorial decision also) is just distracting to readers here.

Reply:
Sadly, it appears to me that we are not covering ground despite my best attempts to outline the, as I believe legitimate, reasoning for thoughtful balance. I also take slight offense by the accusation to contribute to distraction while simultaneously being subjected to distraction from the contents of my comments. Perhaps it makes sense to take a step back at this point, and see if others have opinions that would help to unstuck the arguably unproductive exchange.